firmitas/base/maintool.py

```diff
@@ -54,49 +54,117 @@ def readcert(certobjc):
     return strtdate, stopdate, cstarted, cstopped, daystobt, daystodd, issuauth, serialno
 
 
+def generate():
+    logrdata.logrobjc.info("Generating into the configured directory")
+    doneqant, failqant, totlqant = 0, 0, 0
+
+    logrdata.logrobjc.info("Validating X.509-standard TLS certificates")
+    certloca = Path(standard.certloca)
+
+    for file in certloca.iterdir():
+        if not file.is_file() or ".crt" not in file.name:
+            continue
+
+        certpath = Path(file.as_posix())
+        totlqant += 1
+
+        if not os.path.exists(certpath):
+            logrdata.logrobjc.warning(
+                f"[{file.stem}] The specified X.509-standard TLS certificate could not "
+                + "be located"
+            )
+            failqant += 1
+            continue
+
+        try:
+            readdata = readcert(
+                x509.load_pem_x509_certificate(certpath.read_bytes(), default_backend())
+            )
+            logrdata.logrobjc.info(
+                f"[{file.stem}] The specified X.509-standard TLS certificate was read successfully"
+            )
+            standard.certdict[file.name] = {
+                "path": file.as_posix(),
+                "user": standard.username,
+                "certstat": {
+                    "strtdate": readdata[0],
+                    "stopdate": readdata[1],
+                    "cstarted": readdata[2],
+                    "cstopped": readdata[3],
+                    "daystobt": readdata[4],
+                    "daystodd": readdata[5],
+                    "issuauth": readdata[6],
+                    "serialno": readdata[7],
+                },
+                "notistat": {
+                    "done": False,
+                    "link": "",
+                    "time": "",
+                }
+            }
+            doneqant += 1
+        except ValueError:
+            logrdata.logrobjc.error(
+                f"[{file.stem}] The specified X.509-standard TLS certificate could not be read"
+            )
+            failqant += 1
```

Comment:
The `try:` block is pretty long, it should probably only contain the `readdata = readcert(…)` lines, the rest could go in an `else:` block (i.e. "do this if no exception was raised").